pkg/ebpf/btf.go

```diff
@@ -381,28 +382,20 @@ func (b *orderedBTFLoader) getEmbeddedBTF(platform btfPlatform, platformVersion,
 
 func (b *orderedBTFLoader) searchEmbeddedCollection(filename string) []string {
 	var matchingPaths []string
-	collectionTarball := filepath.Join(b.embeddedDir, "minimized-btfs.tar.xz")
-	err := archiver.NewTarXz().Walk(collectionTarball, func(f archiver.File) error {
-		th, ok := f.Header.(*tar.Header)
-		if !ok {
-			return fmt.Errorf("expected header to be *tar.Header but was %T", f.Header)
-		}
-
-		if !f.IsDir() {
-			if filepath.Base(th.Name) == filename {
-				pform := strings.Split(th.Name, string(os.PathSeparator))[0]
+	collectionTarball := filepath.Join(b.embeddedDir, btfArchiveName)
+	// ignore error because we only care if there are matching paths
+	_ = archive.WalkTarXZArchive(collectionTarball, func(_ *tar.Reader, hdr *tar.Header) error {
+		if hdr.Typeflag&tar.TypeReg != 0 {
```

Comment:
Looking at [the docs](https://pkg.go.dev/archive/tar#pkg-constants), it seems you should use direct comparison here as you do below:

```suggestion
		if hdr.Typeflag == tar.TypeReg {
```

Reply:
Yeah, I was initially confused by the naming of it as flag. Fixed.